lib/posix-process/process.h

```diff
@@ -58,6 +67,7 @@ struct posix_thread {
 	struct uk_list_head thread_list_entry;
 	struct uk_thread *thread;
 	struct uk_alloc *_a;
+	enum posix_thread_state state;
```

Comment:
I would also add a comment here that states again the potential state difference between uk_thread and posix_thread